swap_meet/vendor.py

```diff
@@ -1,2 +1,103 @@
+from inspect import trace
+from .item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    # Additional attribute "trait" is added for the optional swap_by_newest function.
+    # Trait can be Good or Evil, by default it is Good.
+    # When there is an age tie for swap_by_newest, Good venodr swaps the item with the best condition,
+    # While Evil vendor sawps the item with the worst condition.
+    def __init__(self, inventory=None, trait="Good"):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+        self.trait = trait
+
+    def add(self, new_item):
+        self.inventory.append(new_item)
+        return new_item
+    
+    def remove(self, new_item):
+        if new_item in self.inventory:
+            self.inventory.remove(new_item)
+            return new_item
+        return False
+    
+    # WAVE 2
+    def get_by_category(self, category):
+        result = []
+        for item in self.inventory:
+            if item.category == category:
+                result.append(item)
+        return result
+    
+    # WAVE 3
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.add(their_item)
+            other_vendor.add(my_item)
+            self.remove(my_item)
+            other_vendor.remove(their_item)
+            return True
+        return False
+
+    # WAVE 4
+    def swap_first_item(self, other_vendor):
+        if self.inventory and other_vendor.inventory:
+            my_item = self.inventory[0]
+            their_item = other_vendor.inventory[0]
+            self.swap_items(other_vendor, my_item, their_item)
+            return True
+        return False
+
+    # WAVE 6
+    def get_best_by_category(self, category):
+        best_item = None
+    
+        for item in self.inventory:
+            if item.category == category:
+                if best_item == None:
+                    best_item = Item(category=category)
```

Comment:
There is a potential edge case bug here if all of the items of this category have condition 0.  In that case, this function will return this placeholder item rather than an item from the inventory.  An alternative is to initialize best_item with the first item in the list if best_item is None:
```suggestion
                if best_item == None:
                    best_item = item
```